crates/cxx-qt-gen/src/generator/naming/qobject.rs

```diff
@@ -2,64 +2,62 @@
 // SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
 //
 // SPDX-License-Identifier: MIT OR Apache-2.0
-use crate::{generator::naming::CombinedIdent, naming::Name, parser::qobject::ParsedQObject};
+use crate::{
+    naming::{Name, TypeNames},
+    parser::qobject::ParsedQObject,
+};
 use convert_case::{Case, Casing};
 use quote::format_ident;
-use syn::Ident;
+use syn::{Ident, Result};
 
 /// Names for parts of a Q_OBJECT
-pub struct QObjectName {
-    // Store the ident so that cxx_qt_thread_method can use it later
-    ident: Ident,
-    /// The name of the C++ class
-    pub cpp_class: CombinedIdent,
-    /// The name of the Rust struct
-    pub rust_struct: CombinedIdent,
+pub struct QObjectNames {
+    /// The name of the QObject itself.
+    pub name: Name,
+    /// The name of the inner Rust struct
+    pub rust_struct: Name,
     /// The name of the CxxQtThread
     pub cxx_qt_thread_class: Ident,
     /// The name of the Rust closure wrapper to be passed in to CxxQtThread
     pub cxx_qt_thread_queued_fn_struct: Ident,
 }
 
-impl From<&ParsedQObject> for QObjectName {
-    fn from(qobject: &ParsedQObject) -> Self {
-        Self::from_name_and_ident(&qobject.name, qobject.rust_type.clone())
+impl QObjectNames {
+    pub fn from_qobject(qobject: &ParsedQObject, type_names: &TypeNames) -> Result<Self> {
```

Comment:
docs could be useful for these pub methods or make them pub(crate).

Reply:
Done.